cartography/models/core/common.py

```diff
@@ -34,25 +45,51 @@ def __init__(self, name: str, set_in_kwargs=False, extra_index=False, ignore_cas
         """
-        self.name = name
-        self.set_in_kwargs = set_in_kwargs
-        self.extra_index = extra_index
-        self.ignore_case = ignore_case
+        self.name: str = name
+        self.set_in_kwargs: bool = set_in_kwargs
+        self.extra_index: bool = extra_index
+        self.ignore_case: bool = ignore_case
 
     def _parameterize_name(self) -> str:
+        """
+        Private function. Turns self.name into a Neo4j query parameter so that the Neo4j ingestion query can access the
+        value on the dictionary being processed. This is used when parameters are supplied to the querybuilder via
+        kwargs.
+        Relevant Neo4j references: https://neo4j.com/docs/python-manual/current/query-simple/#_write_to_the_database and
+        https://neo4j.com/docs/cypher-manual/current/syntax/parameters/.
+        """
         return f"${self.name}"
 
     def __repr__(self) -> str:
         """
-        `querybuilder.build_ingestion_query()`, generates a Neo4j batched ingestion query of the form
-        `UNWIND $DictList AS item [...]`.
+        Returns string representation of the property so that it can be rendered in the querybuilder based on the
+        various inputs passed to the constructor.
+        """
+        if self.set_in_kwargs:
+            return self._parameterize_name()
 
-        If set_in_kwargs is False (default), we instruct the querybuilder to get the value for this given property from
-        the dict being processed. To do this, this function returns "item.<key on the dict>". This is used for loading
-        in lists of nodes.
+        if self.name.lower() == 'id' or self.ignore_case:
+            # Don't do coalesce() on caseinsensitive attr match.
+            return f"item.{self.name}"
 
-        On the other hand if set_in_kwargs is True, then the value will instead come from kwargs passed to
-        querybuilder.build_ingestion_query(). This is used for things like applying the same update tag to all nodes of
-        a given run.
-        """
-        return f"item.{self.name}" if not self.set_in_kwargs else self._parameterize_name()
+        # Attention: This implementation detail is quirky and deserves the following essay to explain.
+        # This function ensures that the Neo4j query sets the value of this node/relationship property by checking the
+        # following sources in order:
+        # 1. the current dict being processed. That is, we check if dict `item` contains a key called `self.name` with
+        # a non-None value. If so, we are done. Else continue and check the next source.
+        # 2. An existing value on the node itself. That is, we convert self.name to lowercase and check if that is non
+        # null on the Neo4j node `i` already. If None, then the property is not set (this a Neo4j driver behavior).
+        # This means we make an ASSUMPTION that the property name set on the node is the lowercase version of self.name.
```

Comment:
We might need to enforce that PropertyRef variables on NodeSchemas are all lowercase. And also enforce a standard between the variable names and the PropertyRef inputs.

```
myprop = PropertyRef('MyProp')
```

instead of 

```
mycoolprop = PropertyRef('MyProp')
```

Reply:
I agree. Or we need to find another way to allow different intel modules to update the same node types using the same data schema objects. I don't know, can you think of a better way to solve this?

Reply:
For now, another way would be to use subclasses. The ec2/volumes module can have a base version of `EBSVolumeNodeProperties` that does not include `deleteontermination`, while the ec2/instances module has a subclass of `EBSVolumeNodeProperties` that does include that extra property.

```
@dataclass(frozen=True)
class EBSVolumeNodePropertiesFromInstances(EBSVolumeNodeProperties):
    deleteontermination: PropertyRef = PropertyRef('DeleteOnTermination')

@dataclass(frozen=True)
class EBSVolumeSchemaFromInstances(EBSVolumeSchema):
    properties: EBSVolumeNodePropertiesFromInstances = EBSVolumeNodePropertiesFromInstances()
```

This way only the ec2/instances module can overwrite the `deleteontermination` property.

One problem with that: What if we wanted the subclass to have fewer props that the the base class? Then we would need something that could make the `asdict` method preclude the undesired props.